src/Evolution/DgSubcell/Actions/TciAndRollback.hpp

```diff
@@ -135,7 +135,7 @@ struct TciAndRollback {
     const bool bordering_dg_block = alg::any_of(
         element.neighbors(),
         [&subcell_options](const auto& direction_and_neighbor) {
-          const bool first_block_id =
+          const int first_block_id =
```

Comment:
@nilsdeppe This should have been caught by clang-tidy in #4816 but it didn't trigger... looking at the CI check output (https://github.com/sxs-collaboration/spectre/actions/runs/4389805990/jobs/7687734830?pr=4816) it looks like clang-tidy didn't run over this hpp file (maybe because it isn't included in any cpp file? not sure, I haven't looked into clang-tidy for a while). Any idea how we can prevent this sort of bug in the future?

Reply:
I don't know exactly when clang-tidy _should_ check code in an hpp, but I thought it's whenever a cpp file uses that hpp. We run clang-tidy on cpp files only, which is the recommended thing, but a lot of checks that it used and I think should be catching haven't been caught in hpp files. E.g. `const int` declarations in hpp files aren't being flagged at all anymore. At least the local clang-tidy inside my IDE warns about these, so I don't think it's our configuration file. 
From the log it looks like clang-tidy didn't run over any of the `Test_*` files, which if it had I assume it would've noticed this. I don't know why it didn't run over any `Test_*` files, though.